### PR TITLE
Signed CDN urls for backend services

### PR DESCRIPTION
**Pull request by emilymye**
Summary:

- Generate backend services signed url key (tests, docs, etc)
- Add exception to autogenerated example tests template so people don't use autogenerated tests for non-importable resources
- Add signedUrlMaxCacheAgeSecs to BackendService (I expect it will get overriden by https://github.com/GoogleCloudPlatform/magic-modules/pull/1595, or I will remove those files from this PR if they get merged first)

I decided to make this a separate resource from backend_bucket_signed_url_key because I didn't like the alternatives I thought of (enum of "backendBuckets" or "backendServices" to sub into URLs), and because reading them from the backend object might diverge eventually since backend_buckets structure is much more simple than services. 

-----------------------------------------------------------------
# [all]
## [terraform]
Signed CDN urls for backend services
### [terraform-beta]
Signed CDN urls for backend services

### CODE REVIEW

File: third_party/terraform/resources/resource_compute_backend_service.go.erb [hunk 1]
```diff
@@ -92,6 +92,11 @@ func resourceComputeBackendService() *schema.Resource {
 								},
 							},
 						},
+						"signed_url_cache_max_age_sec": {
```
Comment (rileykarson) [inline third_party/terraform/resources/resource_compute_backend_service.go.erb]: drive-by: This file is getting generated by https://github.com/GoogleCloudPlatform/magic-modules/pull/1595, so adding this will be a matter of removing an exclude if you don't mind holding this bit.

Comment (rileykarson) [reply third_party/terraform/resources/resource_compute_backend_service.go.erb]: 1595 is in, so we need to do this through MM now

**Comment by emilymye** [comment]
@rileykarson 👍

File: templates/terraform/examples/base_configs/test_file.go.erb [hunk 2]
```diff
@@ -10,6 +10,8 @@ import (
   "github.com/hashicorp/terraform/helper/resource"
 )
 <%
+raise 'skip_test should be true if resource is not importable' \
```
Comment (chrisst) [inline templates/terraform/examples/base_configs/test_file.go.erb]: What about skipping the import step when generating tests instead? I'm not a fan of 'knowing' which configurations need to be used together and finding out at build time.

File: products/compute/terraform.yaml [hunk 3]
```diff
@@ -234,6 +235,33 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         diff_suppress_func: 'compareSelfLinkOrResourceName'
       timeoutSec: !ruby/object:Overrides::Terraform::PropertyOverride
         default_from_api: true
+  BackendServiceSignedUrlKey: !ruby/object:Overrides::Terraform::ResourceOverride
+    exclude_import: true
+    is_fine_grained_resource: true
```
Comment (chrisst) [inline products/compute/terraform.yaml]: Can this be a property/convenience method based on the existence of `nested_query` rather than its own property?

File: templates/terraform/examples/base_configs/test_file.go.erb (hunk 2 above)
Comment (emilymye) [reply templates/terraform/examples/base_configs/test_file.go.erb]: Import is pretty important for the autogenerated tests since we use it to confirm the attributes. In general, I don't really anticipate exclude_import being used much (i.e. pretty much when the resource contains sensitive information that cannot be returned) and the verification of test resources behavior will probably be unique, so it may not make sense?

File: products/compute/terraform.yaml (hunk 3 above)
Comment (emilymye) [reply products/compute/terraform.yaml]: `:exclude_import` is less because it's fine-grained, more because this has two fields (name and key) and one is sensitive, so it doesn't really make sense to allow for import. Otherwise, we'd have to supply an empty key value, and the user would be asked to apply the key value, which would just recreate the key.

Also, I want to use `nested_query` for collections (i.e. instead of having `self_link_query` for list-only resources) which I do think the current validator code doesn't handle correctly but should include still, vs fine-grained which make less sense to include since they can't be tied to a specific CAI resource.
